### PR TITLE
Implements new indicator CHOP - Choppiness Index

### LINKED ISSUE #8122: Implement Choppiness Index
#### Expected Behavior
Supports Choppiness Index (see [talipp implementation](https://github.com/nardew/talipp/blob/main/talipp/indicators/CHOP.py)).

Refer to [LEAN Engine > Contributions > Indicators](https://www.quantconnect.com/docs/v2/lean-engine/contributions/indicators).

#### Checklist
- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)
- [x] I have provided detailed steps to reproduce the issue

### PR DESCRIPTION
**Pull request by femtotrader**
#### Description

Implement new indicator CHOP - Choppiness Index

#### Related Issue

Closes #8122 

#### Motivation and Context

This indicator wasn't implemented in QuantConnect Lean

#### How Has This Been Tested?

Comparison with talipp computation of CHOP(14) of SPY candles using [generate_reference_data_from_talipp.py](https://github.com/QuantConnect/Lean/blob/master/Tests/TestData/generate_reference_data_from_talipp.py)

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

**Review by Martin-Molinero — CHANGES_REQUESTED** [review]
Thank you 👍 Leaving minor requests

File: Indicators/ChoppinessIndex.cs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+            if (!IsReady)
+            {
+                return 0m;
+            }
+
+            // calculate max high and min low
+            var maxHigh = _highs.Max();
+            var minLow = _lows.Min();
+
+            if (IsReady)
```
[the remark this answers is not in the document]
Comment (femtotrader) [reply Indicators/ChoppinessIndex.cs:97]: Fixed useless IsReady test

CHOP14 is available after 14 incoming data. So I think it's

    WarmUpPeriod= period

True Range return value even after only one incoming data... it's not ready and fully initalized but it return data that is used by RollingWindow (_trueRangeHistory).

File: Indicators/ChoppinessIndex.cs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    /// or not choppy (trading within a trend in either direction)
+    /// </summary>
+    public class ChoppinessIndex : BarIndicator, IIndicatorWarmUpPeriodProvider
+    {
+
+        private readonly int _period;
+        private readonly RollingWindow<float> _highs;
+        private readonly RollingWindow<float> _lows;
+        private readonly IndicatorBase<IBaseDataBar> _trueRange;
+        private readonly RollingWindow<float> _trueRangeHistory;
```
[the remark this answers is not in the document]
Comment (femtotrader) [reply Indicators/ChoppinessIndex.cs]: Fixed by using `decimal` instead

File: Tests/Indicators/ChoppinessIndexTests.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+namespace QuantConnect.Tests.Indicators
+{
+    [TestFixture]
+    public class ChoppinessIndexTests : CommonIndicatorTests<IBaseDataBar>
+    {
+        protected override IndicatorBase<IBaseDataBar> CreateIndicator()
+        {
+            RenkoBarSize = 1m;
+            //VolumeRenkoBarSize = 0.5m;  // AcceptsVolumeRenkoBarsAsInput is hanging when uncommented
```
[the remark this answers is not in the document]
Comment (femtotrader) [reply Tests/Indicators/ChoppinessIndexTests.cs:28]: It really stuck!
I also tried changing VolumeRenkoBarSize to 1m or 2m... but it's still blocking here

**Review by Martin-Molinero — APPROVED** [review]
Nice! 🚀

**Pull request merged**